rpc/client/client_test.go

```diff
@@ -295,6 +295,83 @@ func TestGetBlock(t *testing.T) {
 	require.NoError(err)
 }
 
+func TestValidatedHeight(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+	mockApp, rpc, node := getRPCAndNode(t)
+
+	mockApp.On("BeginBlock", mock.Anything).Return(abci.ResponseBeginBlock{})
+	mockApp.On("Commit", mock.Anything).Return(abci.ResponseCommit{})
+	mockApp.On("InitChain", mock.Anything).Return(abci.ResponseInitChain{})
+
+	err := node.Start()
+	require.NoError(err)
+
+	tests := []struct {
+		name            string
+		validatedHeight uint64
+		nodeHeight      uint64
+		submittedHeight uint64
+		queryHeight     int64
+		result          client.ValidationStatus
+	}{
+		{
```

Comment:
would it make sense to add a test case with `queryHeight: nil` ?

Reply:
test added